### ISSUE TITLE
Add raw HTTP body capability

### ISSUE THREAD

**Issue by clarkington3** [F5Networks/f5-telemetry-streaming#186]
### Is your feature request related to a problem? Please describe.
We are unable to ingest telemetry from Telemetry Streaming because the HTTP body is JSON-formatted.

### Describe the solution you'd like
We would like the ability to have a completely raw HTTP body, giving us total control of how the body is formatted. This will allow much greater flexibility than exists with mandatory JSON formatting.

### Describe alternatives you've considered
The only alternative is to alter the ingestion system to accept JSON formatting however that is not always possible.

### Additional context
By allowing raw HTTP body, you would give complete control of that aspect of the Telemetry Streaming output to users, ensuring maximum compatibility.

In our specific use case the BIG-IP sends data via Telemetry Streaming to the ingestion point however the ingestion point cannot accept JSON-formatted data. Consequently it discards all the traffic received from Telemetry Streaming, even when it is formatted properly, because that payload is then encapsulated in the JSON-formatted mandatory metadata.

We specifically use an Event Listener in TS to receive the data and then the HTTP consumer to transmit the data. It would be ideal if this raw option can be configured at the namespace level or lower (like per event listener or consumer).

**Comment by clarkington3** [comment]
We were able to achieve this functionality by modifying /var/config/rest/iapps/f5-telemetry/lib/consumers/shared/httpUtil.js and adding to the sendToConsumer function. Between the lines "let requestError;" and "return requestsUtil.makeRequest(host, config.uri, requestOptions)" add the following:

```
requestOptions.json = false;
requestOptions.body = requestOptions.body.data;
```

so it looks like:

    let requestError;

    requestOptions.json = false;
    requestOptions.body = requestOptions.body.data;

    return requestsUtil.makeRequest(host, config.uri, requestOptions)

This resulted in the HTTP body of consisting solely of the data sent to the Event Listener, meeting our requirement. This isn't an appropriate long-term solution as it affects all outbound data.

We would be looking for a per-consumer solution, ideally configurable in the consumer. Reviewing requests.js there is a rawResponseBody which defaults to false which might be useful?

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this to our internal product backlog as AUTOTOOL-3037.

**Comment by dstokesf5** [comment]
This issue has been resolved with the 1.29.0 release.
